### DISCUSSION TITLE
In-place skin preview in Settings

### DISCUSSION THREAD

**Discussion by Danielf003** [ppy/osu#19866] [Ideas] [answered]
I'm asking for ability to preview main skin elements for each mode _AT ONCE_ as soon as the skin is selected (or even hovered?). This won't be replaced by the upcoming skin interface unless the latter will not require additional clicks after selecting/hovering the skin in the list. I feel like that _big fullscreen_ interface would be about _movement, gameplay preview, skins creation_ etc., while my _little static_ preview would be for distinguishing skins by _key elements without the need to playtest_. Also this way it's easier to tell which modes are supported by the skin.

Left pic (from Stable) is what inspired me and showed the possibility. General concept is on 2nd and 3rd. I used Aesthetic as an example.
It might look clunky and huge and be costy to performance, that is why there may be an option to hide it (by default) and/or separate modes. Don't look at the o!c fruits too closely, they are just to demonstrate :) Certain skin elements should be fullsize, some can be scaled to save up space. Maybe add or remove some of them, reorder them, hide default (not skinned) parts or change/remove that blue-ish outline - suggestions are welcome, if this is even a viable discussion at all.

Though I'm kinda new to contributing in such big projects, I'd be happy to help if needed and if this idea has any future!

![LazerSkinsPrview](https://user-images.githubusercontent.com/64103844/185714696-73790d97-5fdd-4434-a8b5-d484a8387d0e.png)

**Comment by peppy** [comment, marked as answer]
We're actually making a full-screen skin select with better previews, and hopefully generation of preview images like what you have there (but with better context).
